backend/internal/models/team.go

```diff
@@ -2,14 +2,14 @@ package models
 
 type Team struct {
 	DbModel
-	CorrelationId string `json:"correlation_id"`
-	Name          string `json:"name"`
-	LogoId        string `json:"logo_id"`
-	Color         string `json:"color"`
-	LeagueID      uint   `json:"league_id"`
-	League        League `json:"league"`
-	Roster        Roster `json:"roster"`
-	RosterID      uint   `json:"roster_id"`
+	CorrelationId string  `json:"correlation_id"`
+	Name          string  `json:"name"`
+	LogoId        string  `json:"logo_id"`
+	Color         string  `json:"color"`
+	LeagueID      uint    `json:"league_id"`
+	League        League  `json:"league"`
+	Roster        *Roster `json:"roster"`
+	RosterID      *uint   `json:"roster_id"`
```

Comment:
Can you help me understand why Roster and RosterID require pointers while League doesn't?

Reply:
Because they can be null but a team has to have a league.